test/fuzz/transformation_changing_memory_semantics_test.cpp

```diff
@@ -0,0 +1,479 @@
+               OpFunctionEnd
+  )";
+
+  const auto env = SPV_ENV_UNIVERSAL_1_3;
+  const auto consumer = nullptr;
+  const auto context = BuildModule(env, consumer, shader, kFuzzAssembleOption);
+  spvtools::ValidatorOptions validator_options;
+  ASSERT_TRUE(fuzzerutil::IsValidAndWellFormed(context.get(), validator_options,
+                                               kConsoleMessageConsumer));
+  TransformationContext transformation_context(
+      MakeUnique<FactManager>(context.get()), validator_options);
+  // Bad: Can not be change memory semantics Acquire to Release because it will
+  // change the semantics.
+  ASSERT_FALSE(TransformationChangingMemorySemantics(
+                   MakeInstructionDescriptor(23, SpvOpMemoryBarrier, 0), 0, 27)
+                   .IsApplicable(context.get(), transformation_context));
+  // Bad: Cannot be changed from |UniformMemory| to |WorkgroupMemory|.
+  ASSERT_FALSE(TransformationChangingMemorySemantics(
+                   MakeInstructionDescriptor(23, SpvOpMemoryBarrier, 0), 0, 26)
+                   .IsApplicable(context.get(), transformation_context));
+
+  // Successful transformations.
+  {
+    // Changing memory semantics from |Acquire| to |AcquireRelease| with keeping
+    // the old higher bits value.
+    TransformationChangingMemorySemantics transformation(
+        MakeInstructionDescriptor(23, SpvOpMemoryBarrier, 0), 0, 28);
+    ASSERT_TRUE(
+        transformation.IsApplicable(context.get(), transformation_context));
+    ApplyAndCheckFreshIds(transformation, context.get(),
+                          &transformation_context);
+    ASSERT_TRUE(fuzzerutil::IsValidAndWellFormed(
+        context.get(), validator_options, kConsoleMessageConsumer));
+  }
+
+  std::string after_transformation = R"(
+               OpCapability Shader
+          %1 = OpExtInstImport "GLSL.std.450"
+               OpMemoryModel Logical GLSL450
+               OpEntryPoint GLCompute %4 "main"
+               OpExecutionMode %4 LocalSize 16 1 1
+               OpSource ESSL 320
+          %2 = OpTypeVoid
+          %3 = OpTypeFunction %2
+          %6 = OpTypeInt 32 0
+          %7 = OpConstant %6 2
+         %25 = OpConstant %6 66  ; Acquire | UniformMemory
+         %27 = OpConstant %6 68  ; Release | UniformMemory
+         %26 = OpConstant %6 264 ; AcquireRelease | WorkgroupMemory
+         %28 = OpConstant %6 72  ; AcquireRelease | UniformMemory
+         %20 = OpTypePointer Function %6
+          %4 = OpFunction %2 None %3
+          %5 = OpLabel
+         %21 = OpVariable %20 Function %7
+          %9 = OpFunctionCall %2 %12
+          %8 = OpCopyObject %6 %7
+         %22 = OpLoad %6 %21
+         %23 = OpFunctionCall %2 %12
+               OpMemoryBarrier %7 %28
+         %24 = OpFunctionCall %2 %12
+               OpReturn
+               OpFunctionEnd
+         %12 = OpFunction %2 None %3
+         %13 = OpLabel
+               OpReturn
+               OpFunctionEnd
+  )";
+
```

Comment:
Do the tests all pass?

Reply:
Yes.

Reply:
All tests are passed @paulthomson.

Reply:
@paulthomson All tests are passed in normal build and debug configuration build.